### PR TITLE
feat(tvshowposter): create skeleton loading skeleton for tvshowposter component

### PR DESCRIPTION
**Pull request by simonchen92**
#### Summary

- Work Type: Feature 
- Fixes: 

#### What I did

- Create a skeleton loading UI for tv poster on homepage
-
-

#### Notes / Research / References

-
-

#### Before/After Preview

- Before:

- After:

#### Checklist

- [x] Branch is in format `TYPE/scope/description`
- [x] PR title is in format `TYPE(scope): description`
- [x] Commits are in semantic format
- [x] Has Summary

### CODE REVIEW

**Comment by simonchen92** [comment]
hmm I think something went wrong with this PR...netlify isnt checking the content...

**Comment by davidholyko** [comment]
true... must have something to do with latest eslint / prettier commit hooks.

File: package-lock.json [hunk 1]
```diff
@@ -3992,7 +3992,6 @@
       "version": "2.1.0",
       "resolved": "https://registry.npmjs.org/cli-truncate/-/cli-truncate-2.1.0.tgz",
       "integrity": "sha512-n8fOixwDD6b/ObinzTrp1ZKFzbgvKZvuz/TvejnLn1aQfC6r52XEx85FmuC+3HI+JM7coBRXUvNqEU2PHVrHpg==",
-      "dev": true,
```
Comment (davidholyko) [inline package-lock.json:3995]: Don't commit this package-lock file

File: src/components/Main/TvShowPoster.js [hunk 2]
```diff
@@ -1,14 +1,32 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import { Link } from 'react-router-dom';
+import { Skeleton } from '../../constants/SkeletonLoading';
 
 export const TVShowPoster = ({ data }) => {
+  const [loading, setLoading] = useState(false);
```
Comment (davidholyko) [inline src/components/Main/TvShowPoster.js:6]: I don't think this is the effect we want. This just plays an animation after data has already been loaded.

We want the skeleton to appear before data is fetch and the animation will persist until data is populated

File: src/scss/Skeleton.scss [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    #f1f1f1 40%,
+    #f1f1f1 100%
+  );
+  background-repeat: no-repeat;
+  width: 100%;
+  height: 250px;
+  position: relative;
+}
+
+@keyframes skeletonshimmer {
```
Comment (davidholyko) [inline src/scss/Skeleton.scss:27]: `skeleton-shimmer`

File: src/components/Main/TvShowPoster.js [hunk 4]
```diff
@@ -1,14 +1,40 @@
-import React from 'react';
+import React, { useState } from 'react';
 import { Link } from 'react-router-dom';
+import classNames from 'classnames';
+import { _ } from '../../utils';
 
 export const TVShowPoster = ({ data }) => {
+  const [tvPosterClassNames, setTvPosterClassNames] = useState([
```
Comment (davidholyko) [inline src/components/Main/TvShowPoster.js:7]: we actually dont need any of the logic from here. this file can be reverted and you wouldnt notice a difference

```
git checkout master src/components/Main/TvShowPoster.js
git add 
git commit
```

File: src/scss/Skeleton.scss [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+  height: 250px;
+  margin-top: 20px;
+  margin: auto;
+  background: white;
+  padding: 20px;
+  //   border: 1px solid gray;
+}
+
+.tv-show-poster-image--skeleton {
+  background: linear-gradient(-90deg, #f0f0f0 0%, #f9f9f9 50%, #f0f0f0 100%);
```
Comment (davidholyko) [inline src/scss/Skeleton.scss:12]: the animation is fairly hard to see on against the white background. Maybe more greyer would be better

File: src/components/Main/SkeletonLoading.js [hunk 6]
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
+
+export const Skeleton = () => {
```
Comment (davidholyko) [inline src/components/Main/SkeletonLoading.js:3]: Maybe this file should be TVPosterSkeleton. and changes references as well. We might have more skeletons in the future

File: src/scss/Skeleton.scss (hunk 5 above)
Comment (simonchen92) [reply src/scss/Skeleton.scss:12]: I will do that in my next push

File: src/components/Main/SkeletonLoading.js (hunk 6 above)
Comment (simonchen92) [reply src/components/Main/SkeletonLoading.js:3]: sounds good - I will change the naming convention

File: src/components/Pages/HomePage.js [hunk 7]
```diff
@@ -1,20 +1,21 @@
 import React, { useEffect, useState, useContext } from 'react';
-import { getHighlyRatedEpisodes } from '../../api/api';
+import { AppContext } from '../../contexts/AppContext';
```
Comment (davidholyko) [inline src/components/Pages/HomePage.js:2]: nit sort imports

File: src/components/Pages/HomePage.js [hunk 8]
```diff
@@ -26,10 +27,17 @@ export const HomePage = () => {
...
 
   return (
-    <div>
+    <div className="home-page">
       <SeriesSearchForm />
       <div className="tv-show-poster-section">
-        {_.map(episodesData, (data, index) => (
+        {tvPosterData && !tvPosterData.length && (
+          <React.Fragment>
+            {[...Array(5)].map((item, index) => {
```
Comment (davidholyko) [inline src/components/Pages/HomePage.js:35]: 5 is a fairly arbitrary number. it actually has some semantic meaning. probably make a constant. like `const PREVIEW_SERIES_COUNT = 5` or something. then on line 15, switch the 5 with the variable

File: src/scss/Skeleton.scss [hunk 9]
```diff
@@ -0,0 +1,25 @@
+.tv-show-poster--skeleton {
```
Comment (davidholyko) [inline src/scss/Skeleton.scss:1]: can probably move these to the TVPoster scss

File: src/scss/Skeleton.scss [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+
+.tv-show-poster-image--skeleton {
+  background: linear-gradient(-90deg, #f2f2f2 0%, #d8d8d8 50%, #f2f2f2 100%);
+  animation: skeleton-loading 2s ease-in-out infinite;
+  width: 100%;
+  height: 100%;
+  position: relative;
+}
+
+@keyframes skeleton-loading {
```
Comment (davidholyko) [inline src/scss/Skeleton.scss:18]: move to `Animations.scss`

File: src/components/Pages/HomePage.js [hunk 11]
```diff
@@ -1,20 +1,23 @@
...
+import { getHighlyRatedEpisodes } from '../../api/api';
 import { _ } from '../../utils';
 
 // http://localhost:8080/
 export const HomePage = () => {
   const { setIsLoading } = useContext(AppContext);
-  const [episodesData, setEpisodesData] = useState([]);
+  const [tvPosterData, setTvPosterData] = useState([]);
+
+  const PREVIEW_SERIES_COUNT = 5;
```
Comment (davidholyko) [inline src/components/Pages/HomePage.js:14]: should move to constants 

https://github.com/starter-code/binge/blob/master/src/constants/constants.js

File: src/components/Main/TvPosterSkeletonLoading.js [hunk 12]
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
+
+export const TvPosterSkeleton = () => {
```
Comment (davidholyko) [inline src/components/Main/TvPosterSkeletonLoading.js:3]: nit actually should be TvShowPosterSkeleton

**Review by davidholyko — APPROVED** [review]
LGTM

**Review by davidholyko — APPROVED** [review]
re approved?

**Pull request merged**